lms/djangoapps/course_api/helpers.py

```diff
@@ -53,12 +56,33 @@ def get_marketing_data(course_key, language):
     marketing_root_format = marketing_link('COURSE_DETAILS_API_FORMAT')
     url = marketing_root_format.format(course_id=course_key)
 
-    response = requests.get(url=url, headers={
-        'Accept-Language': language,
-    })
+    try:
+        response = requests.get(
+            url=url,
+            headers={'Accept-Language': language},
+            timeout=settings.EDRAAK_MARKETING_API_TIMEOUT
+        )
+    except requests.exceptions.Timeout:
+        raise APIException(
+            {
+                "status_code": status.HTTP_503_SERVICE_UNAVAILABLE,
+                "developerMessage": "Marketing courses API have timed out."
+            },
+            code=status.HTTP_503_SERVICE_UNAVAILABLE,
+        )
+    except Exception:
+        log.exception('Something went wrong with the marketing courses API')
+        raise APIException(
+            {
+                "status_code": status.HTTP_500_INTERNAL_SERVER_ERROR,
+                "developerMessage": "Marketing courses didn't respond correctly."
```

Comment:
```suggestion
                "developer_message": "Marketing courses didn't respond correctly."
```

Reply:
Consistent with the other platform API. My fault for suggesting the camelcase style.